build_modules/lib/src/module_library.dart

```diff
@@ -105,6 +105,33 @@ class ModuleLibrary {
         (isLibDir && !id.path.startsWith('lib/src/')) || _hasMainMethod(parsed);
     return new ModuleLibrary._fromCompilationUnit(id, isEntryPoint, parsed);
   }
+
+  /// Parses the output of [toString] back into a [ModuleLibrary].
+  ///
+  /// Importable libraries can be round tripped to a String. Non-importable
+  /// libraries should not be printed or parsed.
+  static ModuleLibrary parse(String encoded) {
+    final lines = encoded.split('\n');
+    final id = new AssetId.parse(lines.first);
```

Comment:
We shouldn't need to to encode this in the file, its easily reproducible from the path

Reply:
Done